src/compressor/brotli/BrotliCompressor.cc

```diff
@@ -0,0 +1,86 @@
+#include "brotli/encode.h"
+#include "brotli/decode.h"
+#include "BrotliCompressor.h"
+#include "include/scope_guard.h"
+
+#define MAX_LEN (CEPH_PAGE_SIZE)
+
+int BrotliCompressor::compress(const bufferlist &in, bufferlist &out) 
+{
+  BrotliEncoderState* s = BrotliEncoderCreateInstance(nullptr, nullptr, nullptr);
+  auto sg = make_scope_guard([&s] { BrotliEncoderDestroyInstance(s); });
```

Comment:
https://github.com/ceph/ceph/pull/19549#discussion_r158598352 , please move this after ` if (!s) {}` block.